pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+import "github.com/gogo/protobuf/protobuf/google/protobuf/wrappers.proto";
+
+/*
+IstioInstall is a schema for both defining and overlaying Istio control plane installations.
+The Istio operator uses a compiled-in InstallSpec as default installation values, along with compiled-in install charts.
+Running the operator with an empty user defined InstallSpec results in an installation with default values, using the
+default charts.
+
+The simplest install specialization is to point the user InstallSpec baseValuesPath to a different values file, for
+example an Istio auth installation, which will use the values associated with the auth installation profile for Istio.
+
+Further specialization are possible at three levels:
+
+1. New APIs defined in this file
+
+Feature API: this API groups an Istio install by features and allows enabling/disabling the features, as well as some
+additional high level settings that are feature specific. Each feature contains one or more components, which
+correspond to Istio components in the cluster.
+
+k8s API: this API is a pass through to k8s resource settings for Istio k8s resources. It allows customizing Istio k8s
+resources like Affinity, Resource requests/limits, PodDisruptionBudgetSpec, Selectors etc. in a more consistent and
+k8s specific way compared to values.yaml. See KubernetesResourcesSpec in this file for details.
+
+2. values.yaml
+
+The entirety of values.yaml settings is accessible through InstallSpec. They are included for backward compatibility,
+but also because many of the values will be moved to configuration CRs that are consumed directly by Istio components
+rather than being install time parameters. The only remaining fields will be k8s resource settings which will be
+replaced by the new API above.
+
+Values.yaml profiles are also embedded in the operator, wrapped inside InstallSpecs CommonComponentSpec/ValueOverrides
+field.
+
+3. k8s resource overlays
+
+Once a manifest is rendered from InstallSpec, a further specialization can be applied by specifying k8s resource
+overlays. The concept is similar to kustomize, where JSON patches are applied for object paths. This allows
+customization at the lowest level and eliminates the need to create ad-hoc template parameters.
+
+EXAMPLES
+
+1. Default Istio install
+
+spec: ~
+
+2. Default remote profile install
+
+spec:
+  baseSpecPath: istio-remote.yaml
+
+3. Default install with telemetry disabled
+
+spec:
+  telemetry:
+    enabled: false
+
+4. Default install with each feature installed to different namespace and security components in separate namespaces
+
+spec:
+  traffic_management:
+    namespace: istio-traffic-management
+  policy:
+    namespace: istio-policy
+  telemetry:
+    namespace: istio-telemetry
+  config_management:
+    namespace: istio-config-management
+  security:
+    citadel:
+      namespace: istio-citadel
+    cert_manager:
+      namespace: istio-cert-manager
+    node_agent:
+      namespace: istio-node-agent
+
```

Comment:
This is nice!

Reply:
Thanks Lin. One of the aims with this API is to be super clean and minimal especially for the 80% - if you have any thoughts along those lines, let me know.